ethpm_cli/scraper.py

```diff
@@ -0,0 +1,195 @@
+from collections import OrderedDict
+import itertools
+import json
+from logging import Logger
+from pathlib import Path
+from typing import Any, Dict, Iterable, List, Set, Tuple  # noqa: F401
+
+from eth_utils import to_dict, to_hex, to_int, to_list
+from eth_utils.toolz import assoc
+from ethpm.typing import URI, Address
+from ethpm.utils.backend import resolve_uri_contents
+from ethpm.utils.ipfs import extract_ipfs_path_from_uri, is_ipfs_uri
+from ethpm.utils.uri import is_supported_content_addressed_uri
+from web3 import Web3
+from web3._utils.events import get_event_data
```

Comment:
Noting the import from a private API.  IIUC I think you should be able to define a contract factory that has the relevant event ABI and then use the contract APIs to pull logs that are properly parsed.